lib/vnet/app_resolver.go

```diff
@@ -151,6 +151,21 @@ func (r *TCPAppResolver) ResolveTCPHandler(ctx context.Context, fqdn string) (*T
 	return nil, ErrNoTCPHandler
 }
 
+func (r *TCPAppResolver) fqdnMatchesProfile(ctx context.Context, profileName, fqdn string) bool {
+	if isSubdomain(fqdn, profileName) {
+		return true
+	}
+	// TODO(nklaassen): support leaf clusters.
+	vnetConfig, err := r.clusterConfigCache.getVnetConfig(ctx, profileName, "" /*leafClustername*/)
+	if err != nil {
+		r.slog.DebugContext(ctx, "Failed to get VnetConfig, not checking custom DNS zones.", "profile", profileName, "error", err)
+		return false
+	}
+	return slices.ContainsFunc(vnetConfig.GetSpec().GetCustomDnsZones(), func(zone *vnet.CustomDNSZone) bool {
+		return isSubdomain(fqdn, zone.GetSuffix())
+	})
```

Comment:
Does this mean we're not going to support apps that have `public_addr` set to a second-level domain? I've always assumed that we will. Though now that I look at the RFD, it specifically talks about `*.<custom-dns-zone>`.

Reply:
No it works. I assume we are talking about something like `app.subdomain.customzone.com` in a custom DNS zone of `customzone.com`, that works fine, just added a testcase for it.

Reply:
I was thinking of the app itself having `public_addr` set to `customzone.com`.